gpflux/models/deep_gp.py

```diff
@@ -96,8 +96,8 @@ def __init__(
             If you do not specify a value for this parameter explicitly, it is automatically
             detected from the :attr:`~gpflux.layers.GPLayer.num_data` attribute in the GP layers.
         """
-        self.inputs = tf.keras.Input((input_dim,), name="inputs")
-        self.targets = tf.keras.Input((target_dim,), name="targets")
+        self.inputs = tf.keras.Input((input_dim,), dtype=tf.float64, name="inputs")
+        self.targets = tf.keras.Input((target_dim,), dtype=tf.float64, name="targets")
```

Comment:
this is the main change

Reply:
should we use gpflow default_float() here?

Reply:
probably the default_float() in GPflow makes more sense..so this should be also used in the gpflux.layers.latent_variable_layer

Reply:
yeah, I'll change this bit, thanks